api/v1/controllers/userTokens.js

```diff
@@ -65,63 +65,59 @@ module.exports = {
    */
   deleteUserToken(req, res, next) {
     const resultObj = { reqStartTime: new Date() };
-    authUtils.isAdmin(req)
-    .then((ok) => {
-      if (ok) {
-        const user = req.swagger.params.key.value;
-        const tokenName = req.swagger.params.tokenName.value;
-        const whr = whereClauseForUserAndTokenName(user, tokenName);
-        helper.model.findOne(whr)
-        .then((o) => {
-          if (o) {
-            return o.destroy();
-          }
+    const user = req.swagger.params.key.value;
+    const tokenName = req.swagger.params.tokenName.value;
+    const whr = whereClauseForUserAndTokenName(user, tokenName);
+    let tokenToDel;
 
-          const err = new apiErrors.ResourceNotFoundError();
-          err.resource = helper.model.name;
-          err.key = user + ', ' + tokenName;
-          throw err;
-        })
-        .then((o) => {
-          resultObj.dbTime = new Date() - resultObj.reqStartTime;
-          u.logAPI(req, resultObj, o.dataValues);
-          res.status(httpStatus.OK)
-            .json(u.responsify(o, helper, req.method));
-        })
-        .catch((err) => u.handleError(next, err, helper.modelName));
-      } else {
-        // also OK if user is NOT admin but is deleting own token
-        let userId;
-        authUtils.getUser(req)
-        .then((currentUser) => {
-          userId = currentUser.id;
-          const user = req.swagger.params.key.value;
-          const tokenName = req.swagger.params.tokenName.value;
-          const whr = whereClauseForUserAndTokenName(user, tokenName);
-          helper.model.findOne(whr)
-          .then((o) => {
-            if (o && o.createdBy === userId) {
-              return o.destroy();
-            }
+    // get user token
+    helper.model.findOne(whr)
+    .then((token) => {
+      // if no token found, return error
+      if (!token) {
+        const err = new apiErrors.ResourceNotFoundError();
+        err.resource = helper.model.name;
+        err.key = user + ', ' + tokenName;
+        throw err;
+      }
 
-            const err = new apiErrors.ResourceNotFoundError();
-            err.resource = helper.model.name;
-            err.key = user + ', ' + tokenName;
-            throw err;
-          })
-          .then((o) => {
-            resultObj.dbTime = new Date() - resultObj.reqStartTime;
-            u.logAPI(req, resultObj, o.dataValues);
-            res.status(httpStatus.OK)
-              .json(u.responsify(o, helper, req.method));
-          })
-          .catch((err) => u.handleError(next, err, helper.modelName));
-        });
+      // Default token cannot be deleted
+      if (token.name === token.User.name) {
```

Comment:
Do we create a default token with the same name as the username every time a user is created?